### ISSUE TITLE
[iOS] messaging().getInitialNotification() doesn't work. 

### ISSUE THREAD

**Issue by lazyhoneyant** [invertase/react-native-firebase#517]
When my app is running on background or killed completely, I send a push notification.
iOS shows the notification banner which I sent. 
Then I've touched the banner and awake the app.
At that time, invoked not messaging().getInitialNotification but messaging().onMessage.
Only trigger onMessage from push notification.
To use getInitialNotificaton(), Is there another thing to do?

I put these code in componentDidMount() of root component.

    firebase.messaging().getInitialNotification()
      .then((notification) => {
        console.log('Notification which opened the app: ', notification);
      });

1. Application Target Platform: iOS
2. Development Operating System: macOS Sierra
3. Build Tools: Xcode 9.0
4. React Native version: 0.48 
5. RNFirebase Version: 3.0.2
6. Firebase Module: messaging

**Comment by mayankagarwal2402** [comment]
I'm facing the same issue. This issue should still be open and needs to be fixed either in the documentation or the code.

Although the functionality of `messaging().getInitialNotification()` can be mimiced by using a common handler for both `getInitialNotification()` and `onMessage()` by making use of the `opened_from_tray` flag received as a property of the notification object. 

_Like this..._

```javascript

const onBackgroundMessage = (msg) => {
  // do something with msg
  console.log('Message received in closed app', msg);
}

const onForegroundMessage = (msg) => {
  // do something with msg
  console.log('Message received in open app', msg);
}

const unsubscribeOnMessage = messaging.onMessage(msg => {
  const { opened_from_tray } = msg;

  // opened_from_tray is a numeric boolean and takes values either 1 or 0
  if (opened_from_tray) {
    onBackgroundMessage(msg);
  } else {
    onForegroundMessage(msg);
  }
});

```

**Comment by chrisbianca** [comment]
@mayankagarwal2402 This will all be resolved as part of #595

**Comment by chrisbianca** [comment]
Good news, the long awaited alpha of our messaging and notifications overhaul is now available!!

Check out the release notes here: https://github.com/invertase/react-native-firebase/releases/tag/v4.0.0-alpha.1

If you have any comments and suggestions or want to report an issue, come find us on [Discord](https://discord.gg/C9aK28N)
